src/mesh/replicated_mesh.C

```diff
@@ -24,95 +24,62 @@
-        // Note that this may cause something strange to happen, as Roy
-        // believes he can prove it is not a total ordering...
-        Real rel_size = std::max(std::abs(lhs(i)), std::abs(rhs(i)));
-
-        // Don't use relative tolerance if both numbers are already small.
-        // How small?  Some possible options are:
-        // * std::numeric_limits<Real>::epsilon()
-        // * TOLERANCE
-        // * 1.0
-        // If we use std::numeric_limits<Real>::epsilon(), we'll
-        // do more relative comparisons for small numbers, but
-        // increase the chance for false positives?  If we pick 1.0,
-        // we'll never "increase" the difference between small numbers
-        // in the test below.
-        if (rel_size < 1.)
-          rel_size = 1.;
-
-        // Don't attempt the comparison if lhs(i) and rhs(i) are too close
-        // together.
-        if ( std::abs(lhs(i) - rhs(i)) / rel_size < _tol)
-          continue;
-
-        if (lhs(i) < rhs(i))
-          return true;
-        if (lhs(i) > rhs(i))
-          return false;
-      }
+  VectorOfNodesAdaptor(const std::vector<std::pair<Point, dof_id_type>> & nodes) :
+    _nodes(nodes)
+  {}
+
+  /**
+   * Must return the number of data points
+   */
+  inline size_t kdtree_get_point_count() const { return _nodes.size(); }
+
+  /**
+   * \returns The dim'th component of the idx'th point in the class:
+   * Since this is inlined and the "dim" argument is typically an immediate value, the
+   *  "if's" are actually solved at compile time.
+   */
+  inline Real kdtree_get_pt(const size_t idx, int dim) const
+    {
+      libmesh_assert_less (idx, _nodes.size());
+      libmesh_assert_less (dim, 3);
 
-    // We compared all the components without returning yet, so
-    // each component was neither greater than nor less than they other.
-    // They might be equal, so return false.
-    return false;
-  }
+      const Point & p(_nodes[idx].first);
 
-  // Needed by std::sort on vector<pair<Point,id>>
-  bool operator()(const std::pair<Point, dof_id_type> & lhs,
-                  const std::pair<Point, dof_id_type> & rhs)
-  {
-    return (*this)(lhs.first, rhs.first);
-  }
+      if (dim==0) return p(0);
+      if (dim==1) return p(1);
+      return p(2);
+    }
 
-  // comparison function where lhs is a Point and rhs is a pair<Point,dof_id_type>.
-  // This is used in routines like lower_bound, where a specific value is being
-  // searched for.
-  bool operator()(const Point & lhs, std::pair<Point, dof_id_type> & rhs)
-  {
-    return (*this)(lhs, rhs.first);
-  }
+  /*
+   * Optional bounding-box computation
+   */
+  template <class BBOX>
+  bool kdtree_get_bbox(BBOX & /* bb */) const { return false; }
 
-  // And the other way around...
-  bool operator()(std::pair<Point, dof_id_type> & lhs, const Point & rhs)
-  {
-    return (*this)(lhs.first, rhs);
-  }
 };
-}
 
 
-
-namespace libMesh
-{
+//namespace libMesh
+//{
```

Comment:
Delete these two commented out lines since the namespace declaration has been moved up.